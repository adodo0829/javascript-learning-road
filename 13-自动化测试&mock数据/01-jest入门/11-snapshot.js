
export const generateConfig = () => {
    return {
        server: 'http://localhost',
        port: 8080
    }
}

export const generateConfig2 = () => {
    return {
        server: 'http://localhost',
        port: 8080,
        time: new Date()
    }
}