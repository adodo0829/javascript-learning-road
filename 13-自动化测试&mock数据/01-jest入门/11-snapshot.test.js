/**
 * snapshot 快照测试
 * 每次需要更新配置文件的时候，需要按 u 更新全部快照，按 i 进入交互式单独更新
 */
import { generateConfig, generateConfig2 } from "./11-snapshot";

test("测试 generateConfig 函数", () => {
  expect(generateConfig()).toMatchSnapshot();
});

test("测试 generateConfig2 函数", () => {
  expect(generateConfig2()).toMatchSnapshot({
    // 用于匹配时间类变化的值
    time: expect.any(Date)
  });
});

/**
 * inline snapshot
 * 需要安装 prettier 模块
 * $ npm install prettier --save
 */
// test("测试 generateConfig2 函数 inline", () => {
//     expect(generateConfig2()).toMatchInlineSnapshot({
//         // 用于匹配时间类变化的值
//         time: expect.any(Date)
//     });
// });
test("测试 generateConfig2 函数 inline", () => {
  expect(generateConfig2()).toMatchInlineSnapshot(
    {
      // 用于匹配时间类变化的值
      time: expect.any(Date)
    },
    `
    Object {
      "port": 8080,
      "server": "http://localhost",
      "time": Any<Date>,
    }
  `
  );
});
