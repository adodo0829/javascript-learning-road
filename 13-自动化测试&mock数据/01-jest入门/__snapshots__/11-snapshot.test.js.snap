// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`测试 generateConfig 函数 1`] = `
Object {
  "port": 8080,
  "server": "http://localhost",
}
`;

exports[`测试 generateConfig2 函数 1`] = `
Object {
  "port": 8080,
  "server": "http://localhost",
  "time": Any<Date>,
}
`;
